### PR TITLE
RDoc-2567 + RDoc-2595 Configuration Conventions

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issues:**

https://issues.hibernatingrhinos.com/issue/RDoc-2595/5.4-Add-all-missing-conventions-in-Configuration-Conventions
https://issues.hibernatingrhinos.com/issue/RDoc-2567/Node.js-Client-API-Configuration-Conventions
https://issues.hibernatingrhinos.com/issue/RDoc-1646/Migration-Client-API-Conventions-add-PreserveDocumentPropertiesNotFoundOnModel
https://issues.hibernatingrhinos.com/issue/RDoc-2578/Document-allowing-to-turn-off-writing-raven-cluster-topology-files
https://issues.hibernatingrhinos.com/issue/RDoc-2615/Document-the-HttpVersion-convention

---

**Node.js** - @ml054 - pls review files:
```
Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.js.markdown
Documentation/5.4/Samples/nodejs/client-api/configuration/conventions.js
```
Note: 
* 6.0 files are exactly the same, were just copied over
* I have opened a separate issue for all conventions I need help with (those were Not added in this PR):
  https://issues.hibernatingrhinos.com/issue/RDoc-2613/Node.js-Add-missing-conventions

---

**C#** files to review: @arekpalinski who should review ? you or someone else ?
```
Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown
Documentation/5.4/Samples/csharp/Raven.Documentation.Samples/ClientApi/Configuration/Conventions.cs
Documentation/6.0/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown
Documentation/6.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Configuration/Conventions.cs
```

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 1]
```diff
@@ -1,218 +1,770 @@
...
+{NOTE: }
+
+#### FindClrType
+
+---
+
+* Use the `FindClrType` convention to define a function that finds the CLR type of a document.  
+
+* DEFAULT:  
+  The CLR type is retrieved from the `Raven-Clr-Type` property under the `@metdata` key in the document.
```
Comment (arekpalinski) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:208]: @metdata -> @metadata`

File: Documentation/6.0/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 2]
```diff
@@ -1,226 +1,783 @@
...
+{NOTE: }
+
+#### HttpVersion
+
+---
+
+* Use the `HttpVersion` convention to set the Http version the client will use when communicating  
+  with the server.
+
+* DEFAULT: `System.Net.HttpVersion.Version11` (HTTP 1.1)
```
Comment (arekpalinski) [inline Documentation/6.0/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:413]: In 6.0 we use HTTP2 by default

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 3]
```diff
@@ -1,218 +1,770 @@
...
-| 64-bit | Lower than or equal to 3GB <br> Greater than 3GB and Lower than or equal to 6GB <br> Greater than 6GB | 64MB <br> 128MB <br> 512MB |
-| 32-bit | | 32MB |
+* __Disabling Caching__:  
 
-* Caching is set **per database**.  
-* **Disabling Caching**:  
-  To disable caching globally, set `MaxHttpCacheSize` to zero.
+    * To disable caching globally, set `MaxHttpCacheSize` to zero.  
+  
+    * Note: When caching is disabled, ALL data requests are sent to the server.
```
Comment (arekpalinski) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:460]: Note that requests are _always_ sent anyway. Even with cache enabled. Although then, when sending a request we provide "If-None-Match" header and if the response didn't change the server returns 304 (NotModified) so we get the result from a local cache.

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 4]
```diff
@@ -1,218 +1,770 @@
...
-| Language | Default convention | Example |
-| ------------- | ----- | --- |
-| C# | PascalCase | OrderLines |
-| Java | camelCase | orderLines |
-| JavaScript | camelCase | orderLines |
+* Use the `SendApplicationIdentifier` convention to `true` to enable sending a unique application identifier to the RavenDB Server.
 
-Whatever naming convention a client uses, is by default reflected on the server-side.  
-These defauls can be customized, e.g. to allow inter-language operability.  
+* Setting tp _true_ allows the server to issue performance hint notifications to the client,
```
Comment (arekpalinski) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:621]: Setting tp -> Setting to

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 5]
```diff
@@ -1,218 +1,770 @@
...
-If you override the default `CreateHttpClient` convention we advise 
-that you also set the HTTP client type correctly using the 
-[HttpClientType](../../client-api/configuration/conventions#httpclienttype) convention.  
+
+#### UseOptimisticConcurrency
+
+---
+
+* When setting the `UseOptimisticConcurrency` convention to `true`,  
+  Optimistic Concurrency checks will be applied for all sessions opened from the Document Store.
```
Comment (arekpalinski) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:716]: Maybe link "Optimistic Concurrency" to a dedicated article?

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown (hunk 1 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:208]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown (hunk 4 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:621]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown (hunk 5 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:716]: done, added link

File: Documentation/6.0/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown (hunk 2 above)
Comment (Danielle9897) [reply Documentation/6.0/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:413]: fixed

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown (hunk 3 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:460]: * Thanks for clarifying this about the caching.
  and - if I understand correctly - when Aggressive Caching is used, no requests are sent to the server at all. 
  Instead, the client is notified via the Changes API about any changes.
  
* For now - will remove this problematic note and add links to dedicated articles instead.

**Review by arekpalinski — APPROVED** [review]

File: Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown [hunk 6]
```diff
@@ -1,218 +1,774 @@
...
+{NOTE: }
+
+#### HttpVersion
+
+---
+
+* Use the `HttpVersion` convention to set the Http version the client will use when communicating  
+  with the server.
+
+* DEFAULT: `System.Net.HttpVersion.Version11` (HTTP 1.1)
```
Comment (Danielle9897) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:398]: @ppekrol pls advise
Looking at this comment:  https://github.com/ravendb/ravendb/pull/10042#issue-548002187
Should the above default text change ?

Comment (ppekrol) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/configuration/conventions.dotnet.markdown:398]: we are enforcing that to 1.1, if user will set 'null' during DocumenStore initialization, then it switches to framework default

**Pull request merged**